usl_pipeline/cloud_functions/main.py

```diff
@@ -289,15 +291,51 @@ def _build_feature_matrix_from_archive(
                 continue
 
             name = pathlib.PurePosixPath(member.name).name
+            # TODO: Group logic branch by path (per hazard model)
             if name == file_names.ELEVATION_TIF:
                 return _read_elevation_features(fd)
+            # Handle heat model input files (WPS)
+            elif name.startswith("met_em") and name.endswith(".nc"):
+                return _read_wps_features(fd)
             # TODO: handle additional archive members.
             else:
                 logging.warning(f"Unexpected member name: {name}")
 
     return None, FeatureMetadata()
 
 
+def _read_wps_features(fd: IO[bytes]) -> Tuple[NDArray, FeatureMetadata]:
+    # Ignore type checker error - BytesIO inherits from expected type BufferedIOBase
+    # https://shorturl.at/lk4om
+    with xr.open_dataset(fd) as ds:  # type: ignore
+        feature = ds.data_vars["SNOALB"].values
+        # Drop time axis - each WPS output file will correspond to single time value
+        feature = numpy.squeeze(feature, axis=0)
+        feature = _reorder_wps_spatial_axes(feature)
+        # TODO: Add lat/lon spatial coordinates to array
+
+    # TODO: Write to metastore
+    return feature, FeatureMetadata()
+
+
+def _reorder_wps_spatial_axes(feature: NDArray) -> NDArray:
+    """Reorders the axes for x-lat and y-lon.
+
+    Reorders dimensions within the array to align with the order expected from AtmoML.
+
+    Args:
+      feature: The array containing the feature and its dimensions
+
+    Returns:
+      A new array with its spatial dimension axes reordered.
+    """
+    # TODO: indices of lat lon might not always be the same if there are
+    # additional spatial dims. May also need to read in coords to find out what
```

Comment:
Are you concerned the different layers of the netcdf won't line up? I'd hope the xarray library returns consistent rasters of each layer.

Reply:
From what I've seen - the axis are sometimes ordered differently, but i will need to some more testing. Removing this for now - working on refining this in a separate branch